core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -34,10 +38,28 @@ oppia.factory('UrlService', ['$window', function($window) {
       return urlParts[1] === 'embed';
     },
     getPathname: function() {
-      return $window.location.pathname;
+      return this.getCurrentUrl().pathname;
+    },
+    getParamValuesAsList: function(queryTag) {
+      var url = this.getCurrentUrl().href.slice(
+        this.getCurrentUrl().href.indexOf('?') + 1).split('&');
+      var parentIdArray = [];
+      for (var i = 0; i < url.length; i++) {
+        var urlparam = url[i].split('=');
+        if (urlparam[0] === queryTag) {
+          parentIdArray.push(urlparam[1]);
+        }
+      }
+      return parentIdArray.length > 0 ? parentIdArray : null;
+    },
+    /* Use UrlService.pushParentIdToUrl(url, explorationId); at the state which
+       redirects to push current exploration id to url stack. */
+    addParams: function(url, queryTag, queryValue) {
```

Comment:
fieldName, fieldValue. Also should be addParam() or addField() (singular)

Please also add a comment above this function to clarify that multiple fields can be added with the same name.

Reply:
Done